### ISSUE TITLE
Custom @escape functions

### ISSUE THREAD

**Issue by myaaaaaaaaa** [itchyny/gojq#271]
Something like this?
```go
gojq.Compile(
	query,
	gojq.WithEscaper("goprint", func(x any) string {
		return fmt.Sprint(x)
	}),
)
```

```jq
$ "abcde" | split("") |
	@json    "json:    \(.)",
	@sh      "sh:      \(.)",
	@goprint "goprint: \(.)"

json:    ["a","b","c","d","e"]
sh:      'a' 'b' 'c' 'd' 'e'
goprint: [a b c d e]
```

**Comment by wader** [comment]
For reference, [jaq](https://github.com/01mf02/jaq) implements it like this:
```sh
$ jaq -n 'def @upcase: ascii_upcase; "world" | @upcase "hello \(.)"'
"hello WORLD"
```

**Comment by myaaaaaaaaa** [comment]
> For reference, [jaq](https://github.com/01mf02/jaq) implements it like this:
> ```sh
> $ jaq -n 'def @upcase: ascii_upcase; "world" | @upcase "hello \(.)"'
> "hello WORLD"
> ```

Pretty cool - although at that point, I think we may as well just redefine the `@foo` syntax to be syntactic sugar for a normal function call:

```diff
-"world" | @upcase "hello \(.)"
+"world" | "hello \(. | upcase)"
```

**Comment by wader** [comment]
@myaaaaaaaaa now jqjq does it the same way at jaq 🥳 but even more general would be nice but then would we extend the the standard library with all the existing format functions minus the "@"? 🤔

**Comment by myaaaaaaaaa** [comment]
> but then would we extend the the standard library with all the existing formats as functions minus the "@"? 🤔

Yes, that would be one of the pitfalls of this method - mainly the redundant `text - tostring` and `json - tojson` functions.
